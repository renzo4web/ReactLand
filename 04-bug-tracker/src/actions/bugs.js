import Swal from 'sweetalert2';
import { db } from '../firebase/firebase-config';
import { loadBugs } from '../helpers/loadBugs';
import { types } from '../types/types';

export const addNewBug = (bug) => {
    return async (dispatch, getState) => {
        const { name } = getState().auth;

        const newBug = {
            title: bug.name,
            reporter: name,
            description: bug.description,
            severity: bug.severity,
            assignee: bug.assignee || name,
            status: bug.status,
            createdAt: new Date().getTime(),
        };

        const docRef = await db.collection('bugs').add(newBug);
        try {
            const docAdded = await docRef;
            console.log(docAdded.id);
            await db
                .collection('bugs')
                .doc(docAdded.id)
                .update({ id: docAdded.id });
        } catch (error) {
            console.warn(error);
        }

        Swal.fire('Bug Added', '', 'success');
    };
};

export const startLoadingBugs = () => {
    return async (dispatch) => {
        const bugs = await loadBugs();
        dispatch(setBugs(bugs));
    };
};

export const setBugs = (bugs) => ({
    type: types.bugsLoad,
    payload: bugs,
});

export const startEditBug = (bug) => {
    return async (dispatch) => {
        await db.collection('bugs').doc(bug.id).update(bug);
        Swal.fire('Edited', 'the bug has been correctly edited', 'success');
    };
};

export const deleteBug = (id) => {
    return async (dispatch) => {
        await db.collection('bugs').doc(id).delete();
    };
};
