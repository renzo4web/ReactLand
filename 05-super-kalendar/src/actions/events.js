import toast from 'react-hot-toast';
import { fetchWithToken } from '../helpers/fetch';
import { prepareEvents } from '../helpers/prepareEvents';
import { types } from '../types/types';

/*
    ADDING THE EVENTS TO DB
*/

export const eventStartAddNew = event => {
	return async (dispatch, getState) => {
		const { name, uid } = getState().auth;

		try {
			const response = await fetchWithToken('events', event, 'POST');
			const body = await response.json();

			if (body.ok) {
				dispatch(
					addNewEvent({
						...event,
						id: body.event.id,
						user: {
							_id: uid,
							name,
						},
					})
				);
			} else {
				toast.error(body.msg);
			}
		} catch (error) {
			console.log(error);
			toast.error(error);
		}
	};
};

const addNewEvent = event => ({
	type: types.eventAddNew,
	payload: event,
});

export const setActiveEvent = event => ({
	type: types.eventSetActive,
	payload: event,
});

export const cleanActiveEvent = () => ({
	type: types.eventCleanActive,
});

/*
    UPDATE EVENT
*/
export const eventStartUpdate = event => {
	return async dispatch => {
		try {
			const response = await fetchWithToken(
				`events/${event.id}`,
				event,
				'PUT'
			);

			const body = await response.json();
			if (body.ok) {
				dispatch(updatedEvent(event));
			} else {
				toast.error(body.msg);
			}
		} catch (error) {
			console.log(error);
			toast.error(error);
		}
	};
};

const updatedEvent = event => ({
	type: types.eventUpdated,
	payload: event,
});

/*
    DELETE EVENT
*/

export const eventStartDelete = () => {
	return async (dispatch, getState) => {
		const { activeEvent } = getState().calendar;
		try {
			const response = await fetchWithToken(
				`events/${activeEvent.id}`,
				{},
				'DELETE'
			);

			const { ok, msg } = await response.json();

			if (ok) {
				dispatch(deletedEvent());
			} else {
				toast.error(msg);
			}
		} catch (error) {
			console.log(error);
			toast.error(error);
		}
	};
};

const deletedEvent = () => ({
	type: types.eventDeleted,
});

export const eventsCleanAll = () => ({
	type: types.eventCleanAll,
});

/*
    LOADING THE EVENTS FROM DB
*/

export const eventStartLoading = () => {
	return async dispatch => {
		try {
			const response = await fetchWithToken('events');

			const body = await response.json();

			if (body.ok) {
				const eventsPrepared = prepareEvents(body.events);
				dispatch(eventsLoaded(eventsPrepared));
			} else {
				toast.error(body.msg);
			}
		} catch (error) {
			console.log(error);
		}
	};
};

const eventsLoaded = events => ({
	type: types.eventLoaded,
	payload: events,
});
