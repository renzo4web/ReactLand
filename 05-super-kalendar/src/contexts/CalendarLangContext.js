import React, { createContext, useContext, useState } from 'react';
import { registerLocale, setDefaultLocale } from 'react-datepicker';
import es from 'date-fns/locale/es';

const Context = createContext();

const CalendarLangContext = ({ children }) => {
    const [currLang, setCurrLang] = useState('en');

    if (currLang === 'es') {
        registerLocale('es', es);
        setDefaultLocale('es');
    } else {
        setDefaultLocale('en');
    }

    return (
        <Context.Provider value={{ currLang, setCurrLang }}>
            {children}
        </Context.Provider>
    );
};

export const useCalendarContext = () => {
    return useContext(Context);
};

export default CalendarLangContext;
