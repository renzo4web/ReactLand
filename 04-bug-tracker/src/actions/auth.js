import Swal from 'sweetalert2';

import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { types } from '../types/types';
import { bugsLogout } from './bugs';
import { finishLoading, startLoading } from './ui';

export const startLoginEmailPassword = (email, password) => {
    return (dispatch) => {
        dispatch(startLoading());

        firebase
            .auth()
            .signInWithEmailAndPassword(email, password)
            .then(({ user }) => {
                dispatch(login(user.uid, user.displayName));
                dispatch(finishLoading());
            })
            .catch((err) => {
                dispatch(finishLoading());
                Swal.fire('Error', err.message);
            });
    };
};

export const startGoogleLogin = () => {
    return (dispatch) => {
        firebase
            .auth()
            .signInWithPopup(googleAuthProvider)
            .then(({ user }) => {
                const { displayName, uid } = user;
                dispatch(login(uid, displayName));
            });
    };
};

export const startRegisterWithEmailPasswordName = (name, email, password) => {
    return (dispatch) => {
        firebase
            .auth()
            .createUserWithEmailAndPassword(email, password)
            .then(async ({ user }) => {
                await user.updateProfile({
                    displayName: name,
                });

                dispatch(login(user.id, user.displayName));
            })
            .catch((err) => {
                Swal.fire('Error', err.message);
            });
    };
};

export const login = (uid, displayName) => ({
    type: types.login,
    payload: {
        uid,
        displayName,
    },
});

export const startLogout = () => {
    return async (dispatch) => {
        await firebase.auth().signOut();
        dispatch(bugsLogout());
        dispatch(logout());
    };
};

export const logout = () => ({
    type: types.logout,
});
