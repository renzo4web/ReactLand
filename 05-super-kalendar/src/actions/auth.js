import { types } from '../types/types';
import { fetchWithoutToken, fetchWithToken } from '../helpers/fetch';
import toast from 'react-hot-toast';
import { eventsCleanAll } from './events';

export const startLogin = (email, password) => {
	return async dispatch => {
		// http://localhost:4000/api/auth
		const response = await fetchWithoutToken(
			'auth',
			{ email, password },
			'POST'
		);
		const body = await response.json();

		if (body.ok) {
			localStorage.setItem('token', body.token);
			// keep track initial date of the token created
			localStorage.setItem('token-init-date', new Date().getTime());

			const { uid, name } = body;

			dispatch(
				login({
					uid,
					name,
				})
			);
		} else {
			toast.error(body.msg);
		}
	};
};

export const startRegister = (name, email, password) => {
	return async dispatch => {
		const response = await fetchWithoutToken(
			'auth/new',
			{ email, password, name },
			'POST'
		);

		const body = await response.json();

		if (body.ok) {
			localStorage.setItem('token', body.token);
			// keep track initial date of the token created
			localStorage.setItem('token-init-date', new Date().getTime());

			const { uid, name } = body;

			dispatch(
				login({
					uid,
					name,
				})
			);
		} else {
			toast.error(body.msg);
		}
	};
};

export const startChecking = () => {
	return async dispatch => {
		const response = await fetchWithToken('auth/renew');
		const body = await response.json();

		if (body.ok) {
			localStorage.setItem('token', body.token);
			// keep track initial date of the token created
			localStorage.setItem('token-init-date', new Date().getTime());

			const { uid, name } = body;

			dispatch(
				login({
					uid,
					name,
				})
			);
		} else {
			dispatch(checkingFinish());
		}
	};
};

const checkingFinish = () => ({
	type: types.authCheckingFinish,
});

export const startLogout = () => {
	return dispatch => {
		localStorage.clear();
		dispatch(logout());
		dispatch(eventsCleanAll());
	};
};

const logout = () => ({
	type: types.authLogout,
});

const login = user => ({
	type: types.authLogin,
	payload: user,
});
