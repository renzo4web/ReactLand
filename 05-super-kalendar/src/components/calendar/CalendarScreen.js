import React, { useEffect, useState } from 'react';
import { Calendar, momentLocalizer } from 'react-big-calendar';
import moment from 'moment';
import Container from 'react-bootstrap/Container';

import { useCalendarContext } from '../../contexts/CalendarLangContext';
import { messagesEs } from '../../helpers/calendar-messages-es';
import HeaderNavbar from '../ui/HeaderNavbar';
import CalendarEvent from './CalendarEvent';

import 'moment/locale/es';
import 'react-big-calendar/lib/css/react-big-calendar.css';
import ModalCalendar from './ModalCalendar';
import { useDispatch, useSelector } from 'react-redux';
import {
	cleanActiveEvent,
	setActiveEvent,
	deletedEvent,
	eventStartLoading,
	eventStartDelete,
} from '../../actions/events';
import AddNewFab from '../ui/AddNewFab';
import RemoveEventFab from '../ui/RemoveEventFab';

const localizer = momentLocalizer(moment);

const CalendarScreen = () => {
	const dispatch = useDispatch();
	const { uid } = useSelector(state => state.auth);
	const { events, activeEvent } = useSelector(state => state.calendar);

	const [lastView, setLastView] = useState(
		() => localStorage.getItem('lastView') || 'month'
	);
	const [show, setShow] = useState(false);

	const { currLang } = useCalendarContext();

	useEffect(() => {
		dispatch(eventStartLoading());
	}, []);

	const onDoubleClick = () => {
		setShow(true);
	};

	const onSelect = event => {
		dispatch(setActiveEvent(event));
	};

	const onViewChange = event => {
		setLastView(event);
	};

	const handleSelectSlot = event => {
		dispatch(cleanActiveEvent());
		const { start, end } = event;
		dispatch(setActiveEvent({ start, end }));
		setShow(true);
	};

	const eventStyleGetter = (event, start, end, isSelected) => {
		const style = {
			backgroundColor: uid === event.user._id ? '#367CF7' : '#465660',
			borderRadius: '0px',
			opacity: 0.8,
			display: 'block',
			color: 'white',
		};

		return {
			style,
		};
	};

	moment.locale(currLang);

	return (
		<Container fluid>
			<HeaderNavbar />

			<ModalCalendar show={show} setShow={setShow} />

			<Container className={'my-4'} fluid>
				<Calendar
					localizer={localizer}
					events={events}
					startAccessor="start"
					endAccessor="end"
					style={{ height: '80vh' }}
					eventPropGetter={eventStyleGetter}
					onDoubleClickEvent={onDoubleClick}
					onSelectEvent={onSelect}
					onSelectSlot={handleSelectSlot}
					onView={onViewChange}
					view={lastView}
					selectable
					messages={currLang === 'es' && messagesEs}
					components={{
						event: CalendarEvent,
					}}
				/>
			</Container>

			<AddNewFab
				onClick={() => {
					setShow(true);
					dispatch(cleanActiveEvent());
				}}
			/>

			{activeEvent && (
				<RemoveEventFab onClick={() => dispatch(eventStartDelete())} />
			)}
		</Container>
	);
};

export default CalendarScreen;
