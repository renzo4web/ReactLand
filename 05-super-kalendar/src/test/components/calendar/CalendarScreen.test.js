import { act } from '@testing-library/react';
import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { setActiveEvent } from '../../../actions/events';
import CalendarScreen from '../../../components/calendar/CalendarScreen';
import CalendarLangContext from '../../../contexts/CalendarLangContext';

jest.mock('../../../actions/events', () => ({
	eventStartDelete: jest.fn(),
	setActiveEvent: jest.fn(),
	eventStartLoading: jest.fn(),
}));

Storage.prototype.setItem = jest.fn();

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
	auth: {
		uid: '123123123',
		name: 'test',
	},
	calendar: {
		events: [],
		activeEvent: null,
	},
};
const store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
	<Provider store={store}>
		<CalendarLangContext>
			<CalendarScreen />
		</CalendarLangContext>
	</Provider>
);

describe('Test components/calendar/CalendarScreen', () => {
	test('should render properly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('test Calendar interactions', () => {
		const calendar = wrapper.find('Calendar');

		expect(calendar.exists()).toBe(true);

		calendar.prop('onSelectEvent')();
		expect(setActiveEvent).toHaveBeenCalled();
		act(() => {
			calendar.prop('onView')('week');
		});
		expect(localStorage.setItem).toHaveBeenCalledWith('lastView', 'week');
	});
});
