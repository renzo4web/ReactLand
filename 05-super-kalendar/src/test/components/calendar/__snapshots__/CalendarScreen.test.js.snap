// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test components/calendar/CalendarScreen should render properly 1`] = `
<div
  className="container-fluid"
>
  <header
    className="navbar navbar-expand navbar-dark bg-dark"
    role="navigation"
  >
    <div
      className="py-3 container"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="#home"
        >
          <img
            alt=""
            className="d-inline-block align-top"
            height="30"
            src="https://emojipedia-us.s3.amazonaws.com/source/skype/289/calendar_1f4c5.png"
            width="30"
          />
           
          test
        </a>
        <button
          className="btn-info-outline btn ml-0 mr-auto btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            fill="currentColor"
            height={30}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 640 512"
            width={30}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M152.1 236.2c-3.5-12.1-7.8-33.2-7.8-33.2h-.5s-4.3 21.1-7.8 33.2l-11.1 37.5H163zM616 96H336v320h280c13.3 0 24-10.7 24-24V120c0-13.3-10.7-24-24-24zm-24 120c0 6.6-5.4 12-12 12h-11.4c-6.9 23.6-21.7 47.4-42.7 69.9 8.4 6.4 17.1 12.5 26.1 18 5.5 3.4 7.3 10.5 4.1 16.2l-7.9 13.9c-3.4 5.9-10.9 7.8-16.7 4.3-12.6-7.8-24.5-16.1-35.4-24.9-10.9 8.7-22.7 17.1-35.4 24.9-5.8 3.5-13.3 1.6-16.7-4.3l-7.9-13.9c-3.2-5.6-1.4-12.8 4.2-16.2 9.3-5.7 18-11.7 26.1-18-7.9-8.4-14.9-17-21-25.7-4-5.7-2.2-13.6 3.7-17.1l6.5-3.9 7.3-4.3c5.4-3.2 12.4-1.7 16 3.4 5 7 10.8 14 17.4 20.9 13.5-14.2 23.8-28.9 30-43.2H412c-6.6 0-12-5.4-12-12v-16c0-6.6 5.4-12 12-12h64v-16c0-6.6 5.4-12 12-12h16c6.6 0 12 5.4 12 12v16h64c6.6 0 12 5.4 12 12zM0 120v272c0 13.3 10.7 24 24 24h280V96H24c-13.3 0-24 10.7-24 24zm58.9 216.1L116.4 167c1.7-4.9 6.2-8.1 11.4-8.1h32.5c5.1 0 9.7 3.3 11.4 8.1l57.5 169.1c2.6 7.8-3.1 15.9-11.4 15.9h-22.9a12 12 0 0 1-11.5-8.6l-9.4-31.9h-60.2l-9.1 31.8c-1.5 5.1-6.2 8.7-11.5 8.7H70.3c-8.2 0-14-8.1-11.4-15.9z"
              key="0"
            />
          </svg>
           : 
          en
        </button>
      </div>
      <button
        className="btn-danger btn-outline btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Logout
      </button>
    </div>
  </header>
  <div
    className="my-4 container-fluid"
  >
    <div
      className="rbc-calendar"
      style={
        Object {
          "height": "80vh",
        }
      }
    >
      <div
        className="rbc-toolbar"
      >
        <span
          className="rbc-btn-group"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Today
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Back
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </span>
        <span
          className="rbc-toolbar-label"
        >
          September 2021
        </span>
        <span
          className="rbc-btn-group"
        >
          <button
            className="rbc-active"
            key="month"
            onClick={[Function]}
            type="button"
          >
            Month
          </button>
          <button
            className=""
            key="week"
            onClick={[Function]}
            type="button"
          >
            Week
          </button>
          <button
            className=""
            key="day"
            onClick={[Function]}
            type="button"
          >
            Day
          </button>
          <button
            className=""
            key="agenda"
            onClick={[Function]}
            type="button"
          >
            Agenda
          </button>
        </span>
      </div>
      <div
        aria-label="Month View"
        className="rbc-month-view"
        role="table"
      >
        <div
          className="rbc-row rbc-month-header"
          role="row"
        >
          <div
            className="rbc-header"
            key="header_0"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Sun
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_1"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Mon
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_2"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Tue
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_3"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Wed
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_4"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Thu
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_5"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Fri
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_6"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Sat
            </span>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  31
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  02
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  03
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  04
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-today"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  05
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  06
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  07
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  08
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-now rbc-current"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  09
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  10
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  11
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  12
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  13
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  14
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  15
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  16
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  17
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  18
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  19
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  20
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  21
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  22
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  23
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  24
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  25
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  26
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  27
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  28
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  02
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="rounded-circle p-2 position-fixed bottom-0 end-0 m-xl-4 btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height={50}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width={50}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M16 8A8 8 0 110 8a8 8 0 0116 0zM8.5 4a.5.5 0 00-1 0v3.5H4a.5.5 0 000 1h3.5V12a.5.5 0 001 0V8.5H12a.5.5 0 000-1H8.5V4z"
        fillRule="evenodd"
        key="0"
      />
    </svg>
  </button>
</div>
`;
